File: content/snippet/code-signal/arcade/intro/dark-wilderness/knapsack-light/knapsack-light.py
def knapsack_light(value_1, weight_1, value_2, weight_2, max_weight):
    if weight_1 + weight_2 <= max_weight:
        return value_1 + value_2
    elif weight_1 <= max_weight and weight_2 <= max_weight:
        return max(value_1, value_2)
    elif weight_1 <= max_weight:
        return value_1
    elif weight_2 <= max_weight:
        return value_2
    else:
        return 0


print(knapsack_light(10, 5, 6, 4, 8))

File: content/snippet/code-signal/arcade/intro/dark-wilderness/knapsack-light/knapsack-light.js
function knapsackLight(value1, weight1, value2, weight2, maxW) {
  if (weight1 + weight2 <= maxW) {
    return value1 + value2;
  } else if (weight1 <= maxW && weight2 <= maxW) {
    return Math.max(value1, value2);
  } else if (weight1 <= maxW) {
    return value1;
  } else if (weight2 <= maxW) {
    return value2;
  } else {
    return 0;
  }
}

console.log(knapsackLight(10, 5, 6, 4, 8));
